### PR TITLE
added key based asset transfer status screen and other fixes

### PR DESCRIPTION
**Pull request by poocart**
Added screen to show current status of key based asset transfer transactions and some other fixes/improvements.

### CODE REVIEW

File: src/actions/keyBasedAssetTransferActions.js [hunk 1]
```diff
@@ -299,23 +279,22 @@ export const checkKeyBasedAssetTransferTransactionsAction = () => {
         const transactionSent = await transferSigned(assetToTransferTransaction?.signedHash || '')
           .catch((error) => ({ error }));
         if (!transactionSent?.hash || transactionSent.error) {
-          reportLog('Failed to send key based asset transger signed transaction', {
+          reportLog('Failed to send key based asset transfer signed transaction', {
```
Comment (IAmKio) [inline src/actions/keyBasedAssetTransferActions.js:282]: Good spot! 👍

File: src/screens/KeyBasedAssetTransfer/KeyBasedAssetTransferStatus.js [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+    signedTransaction,
+    transactionHash,
+    amount,
+    status,
+    assetData,
+  }) => ({
+    ...buildHistoryTransaction({
+      from: signedTransaction?.from || '',
+      to: signedTransaction?.to || '',
+      hash: transactionHash || '',
```
Comment (IAmKio) [inline src/screens/KeyBasedAssetTransfer/KeyBasedAssetTransferStatus.js:93]: I'm assuming this ends up being rendered by the UI so hence the `''` - which i can understand

File: src/screens/Menu/Menu.js [hunk 3]
```diff
@@ -18,13 +18,14 @@
     51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 */
 
-import * as React from 'react';
+import React, { useState } from 'react';
```
Comment (IAmKio) [inline src/screens/Menu/Menu.js:21]: Nice optimisation 👍

File: src/screens/Menu/Menu.js [hunk 4]
```diff
@@ -130,98 +132,107 @@ const LockScreenTextLink = styled(TextLink)`
...
+  theme,
+  navigation,
+  goToInvitationFlow,
+  isPillarRewardCampaignActive,
+  hasKeyBasedAssetsTransferInProgress,
+  logoutUser,
+  backupStatus,
+  lockScreen,
+}: Props) => {
+  const [visibleModal, setVisibleModal] = useState(null);
```
Comment (IAmKio) [inline src/screens/Menu/Menu.js:145]: Thank you for the `null` 👍

File: src/screens/KeyBasedAssetTransfer/KeyBasedAssetTransferStatus.js (hunk 2 above)
Comment (poocart) [reply src/screens/KeyBasedAssetTransfer/KeyBasedAssetTransferStatus.js:93]: Yes, this is added to not crash the UI if somehow values didn't make up to here, however, we can logs elsewhere for such cases.

**Pull request merged**